wallet/src/repository/wallet/mod.rs

```diff
@@ -372,8 +364,24 @@ where
             }
         }
 
+        // Append balance movements of pending blocks
+        state.pending_movements.values().for_each(|x| {
+            keyed_transactions.extend(x.iter().map(|movement| (movement.db_key, movement.clone())));
+        });
+
+        let total_txns = keyed_transactions.len();
```

Comment:
Careful, this is number of transactions in the `BTreeMap`, I think you want the index of the highest transaction. Try something like this:

```suggestion
        // Get the highest index
        let last_index = keyed_transactions.values().rev().next().unwrap_or(0);
```

And later when you do

```
total_txns + index
```

It should be

```
total_txns + index + 1
```

To test this we need a `keyed_transactions` where the first key is not 0.